src/main/java/reposense/frontend/RepoSense.java

```diff
@@ -5,41 +5,26 @@
 import java.util.List;
 
 import reposense.dataobject.RepoConfiguration;
+import reposense.exception.ParseException;
+import reposense.parser.ArgsParser;
 import reposense.report.RepoInfoFileGenerator;
 import reposense.system.CsvConfigurationBuilder;
 
 public class RepoSense {
 
     public static void main(String[] args) {
-        if (args.length == 0) {
-            showHelpMessage();
-        }
-
         try {
-            CliArguments cliArguments = new CliArguments(args);
+            CliArguments cliArguments = ArgsParser.parse(args);
 
             Path configFile = cliArguments.getConfigFilePath();
-            Path targetFile = cliArguments.getTargetFilePath();
+            Path targetFile = cliArguments.getOutputFilePath();
             Date fromDate = cliArguments.getSinceDate().orElse(null);
             Date toDate = cliArguments.getUntilDate().orElse(null);
 
             List<RepoConfiguration> configs = CsvConfigurationBuilder.buildConfigs(configFile, fromDate, toDate);
             RepoInfoFileGenerator.generateReposReport(configs, targetFile.toAbsolutePath().toString());
-        } catch (IllegalArgumentException ex) {
-            System.out.print(ex.getMessage());
-            showHelpMessage();
+        } catch (ParseException pe) {
+            System.out.print(pe.getMessage());
```

Comment:
one more minor fix - due to the recent logger addition, please change this to use logger